### PR TITLE
Updating crt/crl files due to expiry before 2027-01-01

### PR DESCRIPTION
**Pull request by lpy4105**
## Description
We have already got https://github.com/Mbed-TLS/mbedtls/pull/7567 tested. It renews all the files that will expire after `2027-01-01`. This PR, which is splitted from it, contains the files that will expire after `2024-01-01` and before `2027-01-01`.

Preceding: #7643

~~The first commit is after https://github.com/Mbed-TLS/mbedtls/pull/7662/commits/686a2a314673c947525e84644b3838d3a4f38a6e~~

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required (test data update)
- [x] **backport**: https://github.com/Mbed-TLS/mbedtls/pull/7768
- [x] **tests** not required

### CODE REVIEW

File: tests/data_files/Makefile [hunk 1]
```diff
@@ -425,9 +460,21 @@ all_final += server5-ss-expired.crt
 
 # try to forge a copy of test-int-ca3 with different key
 server5-ss-forgeca.crt: server5.key
-	$(FAKETIME) '2015-09-01 14:08:43' $(OPENSSL) req -x509 -new -subj "/C=UK/O=mbed TLS/CN=mbed TLS Test intermediate CA 3" -set_serial 77 -config $(test_ca_config_file) -extensions noext_ca -days 3650 -sha256 -key $< -out $@
+	$(OPENSSL) req -x509 -new -subj "/C=UK/O=mbed TLS/CN=mbed TLS Test intermediate CA 3" -set_serial 77 -config $(test_ca_config_file) -extensions noext_ca -days 3650 -sha256 -key $< -out $@
```
Comment (yuhaoth) [inline tests/data_files/Makefile:463]: Nitpick: too long

**Review by yuhaoth — CHANGES_REQUESTED** [review]
Found a wrong typo.

**Review by yuhaoth — DISMISSED** [review]
LGTM

**Comment by lpy4105** [comment]
Rebase to resolve conflict.

**Review by yuhaoth — APPROVED** [review]
LGTM

File: tests/data_files/test-ca2.ku-crt_crl.crt.openssl.v3_ext [hunk 2]
```diff
@@ -0,0 +1,4 @@
+basicConstraints = CA:true
+subjectKeyIdentifier=none
```
Comment (gowthamsk-arm) [inline tests/data_files/test-ca2.ku-crt_crl.crt.openssl.v3_ext:2]: Maybe there is some config that I'm missing, but is `none` accepted? When I try to generate the certificate with `make`, openssl throws the following error.
```
Error Loading extension section default
139977640584896:error:22071071:X509 V3 routines:string_to_hex:illegal hex digit:v3_utl.c:498:
139977640584896:error:22098080:X509 V3 routines:X509V3_EXT_nconf:error in extension:v3_conf.c:95:name=subjectKeyIdentifier, value=none
```

Comment (gilles-peskine-arm) [reply tests/data_files/test-ca2.ku-crt_crl.crt.openssl.v3_ext:2]: I can reproduce this failure with openssl 1.1.1f but not with 3.0.2.

I think the makefile already has commands that require openssl ≥3 and others that don't work with openssl ≥3, so resolving this is out of scope here.

**Review by gowthamsk-arm — APPROVED** [review]

**Pull request merged**